tests/test_mmd.py

```diff
@@ -10,6 +10,7 @@
 METRICS_CONFIG_PATH = os.path.join(PROJECT_ROOT, "tests", "testconfig", "metrics.yaml")
 CONFIGS = load_configs(METRICS_CONFIG_PATH)
 METRIC_CONFIG = CONFIGS["metric_config"]
+METRIC_CONFIG = {k: v for k, v in METRIC_CONFIG.items() if v["class"] == "mmd"}
```

Comment:
Your understanding is correct. There are other options rather than filtering:

1. We could just have one `test_metrics.py` file which would run the tests for all metrics. However, I think having separate scripts for each metric is quite useful for debugging, e.g. future development could cause just _one_ metric to fail in which case it's useful to have a test file to run that single metric while debugging.
2. We could split the `metrics.yaml` file into one file for each metric, i.e. `metrics_mmd.yaml` and `metrics_otdd.yaml`. But I thought it might be useful to see all the test params in one file. 

I didn't much like either of the above options and will implement the setup method as you have suggested.

Reply:
As there is nothing to teardown, I think the fixture can `return` rather than `yield` the dictionary?